.gitignore

```diff
@@ -1,23 +1,451 @@
-# See https://help.github.com/ignore-files/ for more about ignoring files.
+# moban hashes
+.moban.hashes
 
-# compiled output
-/dist
-/tmp
+# Patch & backup files
+*.patch
+*.orig
+*.diff
 
-# dependencies
-/node_modules
-/bower_components
+# Extra rules from https://github.com/github/gitignore/
+# Node rules
+# Logs
+logs
+*.log
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
+
+# Runtime data
+pids
+*.pid
+*.seed
+*.pid.lock
+
+# Directory for instrumented libs generated by jscoverage/JSCover
+lib-cov
+
+# Coverage directory used by tools like istanbul
+coverage
+
+# nyc test coverage
+.nyc_output
+
+# Grunt intermediate storage (http://gruntjs.com/creating-plugins#storing-task-files)
+.grunt
+
+# Bower dependency directory (https://bower.io/)
+bower_components
+
+# node-waf configuration
+.lock-wscript
+
+# Compiled binary addons (https://nodejs.org/api/addons.html)
+build/Release
+
+# Dependency directories
+node_modules/
+jspm_packages/
+
+# TypeScript v1 declaration files
+typings/
+
+# Optional npm cache directory
+.npm
+
+# Optional eslint cache
+.eslintcache
 
+# Optional REPL history
+.node_repl_history
+
+# Output of 'npm pack'
+*.tgz
+
+# Yarn Integrity file
+.yarn-integrity
+
+# dotenv environment variables file
+.env
+
+# next.js build output
+.next
+
+# vuepress build output
+.vuepress/dist
+
+# Serverless directories
+.serverless
+# Sass rules
+.sass-cache/
+*.css.map
+*.sass.map
+*.scss.map
+
+# Emberjs rules
 # misc
-/.sass-cache
 /connect.lock
-/coverage/*
 /libpeerconnection.log
-npm-debug.log*
-yarn-error.log
 testem.log
 
 # ember-try
 .node_modules.ember-try/
 bower.json.ember-try
 package.json.ember-try
+
+
+# VirtualEnv rules
+# Virtualenv
+# http://iamzed.com/2009/05/07/a-primer-on-virtualenv/
+.Python
+[Bb]in
+[Ii]nclude
+[Ll]ib
+[Ll]ib64
+[Ll]ocal
+[Ss]cripts
```

Comment:
Why we want to ignore this in a JS project?

Reply:
Consistency.

Reply:
Well, I thought it is only required in python project.

Reply:
It is required in a python project, but it comes from the [Global VirtualEnv file](https://github.com/github/gitignore/blob/master/Global/VirtualEnv.gitignore), which is among a long list of Global files which are being included so that all repos support all editors/OS/etc.  https://gitlab.com/coala/mobans/merge_requests/27/diffs#5042e0fe174c9e90b403cf80175c9c8b18257fb1_0_3

Then if someone wants to add any more personally important gitignores, we can tell them to go upstream and then we will import them to all repos consistently.